hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileReader.java

```diff
@@ -343,16 +344,18 @@ private static GenericRecord deserialize(final byte[] keyBytes,
                                            final byte[] valueBytes,
                                            Schema writerSchema,
                                            Schema readerSchema) throws IOException {
-    GenericRecord record = HoodieAvroUtils.bytesToAvro(valueBytes, writerSchema, readerSchema);
+    GenericRecord wRecord = HoodieAvroUtils.bytesToAvro(valueBytes, writerSchema);
+    GenericData.Record rRecord = new GenericData.Record(readerSchema);
```

Comment:
I don't think this is the right place to implement this -- we need to be handle schema evolution holistically, not just for HFile reader